scripts/single-choice-set.js

```diff
@@ -833,11 +833,13 @@ H5P.SingleChoiceSet = (function ($, UI, Question, SingleChoice, SolutionView, Re
    * @returns {object}
    */
   SingleChoiceSet.prototype.getCurrentState = function () {
-    return {
-      progress: this.currentIndex,
-      answers: this.results,
-      userResponses: this.userResponses
-    };
+    if (this.userResponses.length > 0) {
+      return {
+        progress: this.currentIndex,
+        answers: this.results,
+        userResponses: this.userResponses
+      };
+    } else return;
```

Comment:
The proposed solution with returning undefined if nothing has been done seems fine, though I would suggest a few slight improvements for better style and readability:
```suggestion
    return this.userResponses.length > 0
      ? {
        progress: this.currentIndex,
        answers: this.results,
        userResponses: this.userResponses
      }
      : undefined;
```